python/rmm/rmm/pylibrmm/stream.pyx

```diff
@@ -34,19 +34,38 @@ cdef class Stream:
         ----------
         obj: optional
             * If None (the default), a new CUDA stream is created.
+            * If a stream that implements the __cuda_stream__ protocol
+              is provided, we use it.
             * If a Numba or CuPy stream is provided, we make a thin
               wrapper around it.
         """
         if obj is None:
             self._init_with_new_cuda_stream()
-        elif isinstance(obj, Stream):
-            self._init_from_stream(obj)
+            return
+        elif hasattr(obj, "__cuda_stream__"):
+            protocol = getattr(obj, "__cuda_stream__")
+            if protocol[0] != 0:
+                raise ValueError("Only protocol version 0 is supported")
+            self._cuda_stream = <cudaStream_t>obj
+            self.owner = obj
         else:
+            # TODO: Remove this branch when numba and cupy
+            # streams implement __cuda_stream__
```

Comment:
For the immediate term, let's remove this code path and focus on just making the current stream APIs public in this PR.

The `__cuda_stream__` protocol [may change its API](https://github.com/NVIDIA/cuda-python/issues/348), and I don't know of any implementations in common libraries like Numba or CuPy that would be able to actively use this. Also we don't have tests written yet. We should definitely add `__cuda_stream__` protocol support -- just in a follow-up PR, after that issue 348 is resolved. That way we can focus on getting the interoperability to be as clear as possible, and add all the tests we need.

Reverting to the previous state here:
```suggestion
        elif isinstance(obj, Stream):
            self._init_from_stream(obj)
        else:
```